internal/state/conditions/httproute.go

```diff
@@ -0,0 +1,72 @@
+package conditions
+
+import (
+	"fmt"
+
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+	"sigs.k8s.io/gateway-api/apis/v1beta1"
+)
+
+// RouteCondition defines a condition to be reported in the status of an HTTPRoute.
+type RouteCondition struct {
+	Type    v1beta1.RouteConditionType
+	Status  metav1.ConditionStatus
+	Reason  v1beta1.RouteConditionReason
+	Message string
+}
+
+// DeduplicateRouteConditions removes duplicate conditions based on the condition type.
+// The last condition wins.
+func DeduplicateRouteConditions(conds []RouteCondition) []RouteCondition {
+	uniqueConds := make(map[v1beta1.RouteConditionType]RouteCondition)
+
+	for _, cond := range conds {
+		uniqueConds[cond.Type] = cond
```

Comment:
the condition time is set in a different place here https://github.com/nginxinc/nginx-kubernetes-gateway/blob/4c82cc02aa50f7bd2da24e02399c610c011c421f/internal/status/httproute.go#L45 when we generate the status of the resource to be reported. we use the time of the moment just before generating and reporting the status

Reply:
How could we have duplicates to begin with? Can duplicates be set to the API server and communicated back to us? Or is this a bug in our processing?

Reply:
Multiple errors can result into the same condition (same type). Right now it is not obvious, because there is still a lot of implement in the status.

However, here is an example: type=Accepted status=False  with reason `UnsupportedValue ` can occur multiple times in one HTTPRoute (it occurs when when certain fields include unsupported values).
 On top of that, type=Accepted status=False with reason `NoMatchingListenerHostname` can occur if the same resource can't find a matching listener. 
So processing a single HTTPRoute can result into having multiple type=Accepted conditions.

However, the HTTPRoute status must include only one per type. This is also enforced by the CRD scheme. So that's why we deduplicate.

Reply:
Will the chosen condition always be the same? Since we're building from a list, if the list order is deterministic then the answer is yes. But will that list order be predictable?

If the condition chosen can change between evaluations I don't think that's good for users.

Reply:
the list order should be deterministic -- because we process things in order. For example, we process routing rules one by one (list of rules), we process parentRefs (list) in a route one by one.

At the same time, we also process things from maps. For example,  here https://github.com/nginxinc/nginx-kubernetes-gateway/blob/cf204aa0712e04e17ed6007454a39239b9fd2fe4/internal/state/graph.go#L81 . However,  I don't see any conditions generated from that.